thumbor/handlers/__init__.py

```diff
@@ -281,7 +281,7 @@ def finish_request(self, context, result_from_storage=None):
             context.config.RESULT_STORAGE_STORES_UNSAFE or not context.request.unsafe)
 
         def inner(future):
-            results, content_type = future.result()
+            results, content_type = (future.result()[0], BaseEngine.get_mimetype(future.result()[0]))
```

Comment:
For the "auto" plugin, I have `'auto' in self.context.request.filters` in the `should_run` call, so it's somewhat different that just checking the `context.request.format`